<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01//EN"
"http://www.w3.org/TR/html4/strict.dtd">
<html>
	<head>
		<title>miRDeep2 tutorial</title>
<style>

h1 {
	    color: #6495ed;
}
.command {
	color: green;
	padding-top:0.5cm;
	padding-bottom:0.5cm;
}

</style>
	</head>
	<body>
		<h1>Disclaimer</h1>
		This tutorial comes with no warranty and demands common sense of the reader. I am not responsible for any damage that happens to your computer by using this tutorial. 
	<h1>Preface</h1>
	This is a step by step guide for a full small RNA sequencing data analysis using the miRDeep2 package and its patched files. The first part will describe the general workflow to do de-novo miRNA predictions based
	on a small RNAseq data seq and the second part will focus on expression analysis with the quantification module. 
	<h1> Installation instructions</h1>
	If you haven't installed them yet you can obtain the main package from 
	<a href="https://github.com/rajewsky-lab/mirdeep2">miRDeep2</a>
	and the patched files from 
	<a href="https://github.com/Drmirdeep/mirdeep2_patch">patch</a>		
	by clicking on 'Clone or download' and then on 'Download Zip'.
	
	Extract the zipped files and then open a command line window.

	If you have git installed you can obtain the packages also directly from the
	command line by typing 
	<div class="command">git clone https://github.com/rajewsky-lab/mirdeep2.git</div>
	and
	<div class="command">git clone https://github.com/Drmirdeep/mirdeep2_patch.git</div>
	To install the miRDeep2 package enter the directory to which the package was extracted to. If you extracted the folder on the Desktop then typing 
	<div class="command">cd ~/Desktop/mirdeep2</div> 
will bring you to the mirdeep2 folder.
Then typing 
<div class="command">perl install.pl</div>
will start the installer and download and install third party software. In particular. bowtie version 1, RNAfold, randfold and the perl packages PDF:API and TTF will be installed. Please follow the instructions on the screen. 

When mirdeep2 was installed successful then you can enter the directory of the mirdeep2_patch 
by typing
<div class="command">cd ~/Desktop/mirdeep2_patch</div>
if you extracted the patched file to the Desktop 
and then just
<div class="command">bash patchme.sh</div>
This will add the patched files to your mirdeep2 installation. 
After that we can start with the tutorial.


	Download the miRBase reference files for version 21 by typing
	<div class="command">mirbase.pl 21</div>
	This will download the hairpin.fa.gz and mature.fa.gz for version 21 to directory
	~/mirbase/21/
		
	If you want the gff files as well then you need to type 
	<div class="command">mirbase.pl 21 1</div>
	The second argument can be anything but 0 which will tell the script to also get the gff files from mirbase. 
	<h1>The data and what else you will need</h1>
	Usually you will have gotten a small RNA sequencing data file from a collaborator that wants you 
	to analyze the data file. Before you can start with any kind of analysis you should either already know
	the small RNA sequencing adapter that was used for the sequencing of the sample or ask your collaborator 
	to sent it to you. If you don't clip the adapter then the majority of the reads having an adapter 
	are likely not to be aligned to anywhere.
	<br><br>
	Once you know the adapter sequence you should do a simple check to see how many 
	of your sequences contain the adapter. This you can do by typing
	<div class="command">grep -c TGGAATTC example_small_rna_file.fa</div>
	where 'TGGAATTC' are the first 8 nucleotides of the adapter that has been used
	for this sample. Replace it with your own sequencing adapter. MicroRNAs have a mean length
	of 22 nucleotides in animals so if you have sequenced one of those it will likely have the 
	sequencing adapter attached to it. If the resulting number of sequences with an adapter is around
	70\% of the number of your input sequences the data set can be considered as reasonably good. 
	If it is only 10\% then likely something went wrong during the sequnecing library preparation or
	or your sample doesn't so many contain small RNAs. 
	
	<h1>Data preprocessing</h1>
	Since the miRDeep2 package was designed as a complete 
	solution for miRNA prediciton and quantification it also contains also data preprocessing routines 
	than will also clip the sequencing adapter. All the data preprocessing as well 
	as the mapping of the preprocessed reads to a reference database. The reference database
	is typically an annotated genome sequence but can also be simply a scaffold assembly if no genome is available.
	The scaffolds itself however should be at least 200 nucleotides long so that a sane miRNA precursor plus some flanking region fits into it. Apart from clipping adapters the module does sanity checks on your 
	sequencing reads and also collapse read sequences to reduce the file size which will save computing time. 

	<div class="command">mapper.pl example_small_rna_file.fastq -e -h -i -j -k TGGAATTC -l 18 -m -p reference_genome_index_bowtie1 -s reads_collapsed.fa -t reads_vs_genome.fa -v -o 4</div>
	What does this command do.
	The first argument needs to be your sequencing file. Typically, this will be a fastq file. The format of the fastq file is designated by specifying option '-e'. If your file is in fasta format already you specify option '-c' instead. If your reads file is not in fasta format you need to specify option '-h' which advises the mapper module to parse your file to fasta format. Option -i will convert RNA to DNA and option '-j' will remove sequences that contain characters other than ACGTN. 
	<br><br>
Now comes the actual adapter clipping which is only done if a adapter sequence is given by option -k. Only the first 6 nucleotides of this sequence will be used to search for an exact match in the sequencing reads. 
Option '-m' will collapse the reads to remove redundancy and decrease the file size. A sequnecing read seen 10 times in your raw file will occur only once in the collapsed file and have a _x10 in its identifier. 
<br><br>
After that the reads will be mapped to the given refence genome which index file was specified by option '-p'.
Option -s indicates the preprocessed read file name which is output by the mapper module and option -t is the file name of the read mappings to the reference genome in miRDeep2's arf format. 
	</body>
</html>
